arch/risc-v/src/common/riscv_backtrace.c

```diff
@@ -162,9 +162,9 @@ int up_backtrace(struct tcb_s *tcb, void **buffer, int size, int skip)
             {
               ret += backtrace(rtcb->stack_base_ptr,
                                rtcb->stack_base_ptr + rtcb->adj_stack_size,
-                               (void *)up_current_regs()[REG_FP],
-                               (void *)up_current_regs()[REG_EPC],
-                               &buffer[ret], size - ret, &skip);
+                               (void *)((uintreg_t *)running_regs())[REG_FP],
```

Comment:
Why is the intermediate cast to uintreg_t * needed ?

Reply:
![image](https://github.com/user-attachments/assets/3584ae40-69c1-461e-9237-ef8a203ade08)
running_regs return (void **) if According to the index, addressing may go wrong